GNU Hurd compilation failure: expected `SystemTime`, found `timespec`

Compiling project with `cargo build -Z build-std=std --target=i686-unknown-hurd-gnu --features="std"` causes the following compilation error:
```
error[E0308]: mismatched types
   --> /home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/pal/unix/fs.rs:520:29
    |
520 |         Ok(SystemTime::from(self.stat.st_mtim))
    |            ---------------- ^^^^^^^^^^^^^^^^^ expected `SystemTime`, found `timespec`
    |            |
    |            arguments to this function are incorrect
    |
note: associated function defined here
   --> /home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/convert/mod.rs:585:8
    |
585 |     fn from(value: T) -> Self;
    |        ^^^^

error[E0308]: mismatched types
   --> /home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/pal/unix/fs.rs:548:29
    |
548 |         Ok(SystemTime::from(self.stat.st_atim))
    |            ---------------- ^^^^^^^^^^^^^^^^^ expected `SystemTime`, found `timespec`
    |            |
    |            arguments to this function are incorrect
    |
note: associated function defined here
   --> /home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/convert/mod.rs:585:8
    |
585 |     fn from(value: T) -> Self;
    |        ^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `std` (lib) due to 2 previous errors
Error: Process completed with exit code 101.
```
See this [CI job](https://github.com/rust-random/getrandom/actions/runs/8390308799/job/23052022971).

--- a/library/std/src/sys/pal/unix/fs.rs
+++ b/library/std/src/sys/pal/unix/fs.rs
@@ -517,7 +517,7 @@ impl FileAttr {
 
     #[cfg(any(target_os = "horizon", target_os = "hurd"))]
     pub fn modified(&self) -> io::Result<SystemTime> {
-        Ok(SystemTime::from(self.stat.st_mtim))
+        SystemTime::new(self.stat.st_mtim.tv_sec as i64, self.stat.st_mtim.tv_nsec as i64)
     }
 
     #[cfg(not(any(
@@ -545,7 +545,7 @@ impl FileAttr {
 
     #[cfg(any(target_os = "horizon", target_os = "hurd"))]
     pub fn accessed(&self) -> io::Result<SystemTime> {
-        Ok(SystemTime::from(self.stat.st_atim))
+        SystemTime::new(self.stat.st_atim.tv_sec as i64, self.stat.st_atim.tv_nsec as i64)
     }
 
     #[cfg(any(

